File: java/src/main/java/club/marzipan/javacarrentals/AccountMapper.java
package club.marzipan.javacarrentals;

import org.springframework.stereotype.Component;

@Component
public class AccountMapper {
    public Account mapEntityToApi(AccountEntity accountEntity) {
        if (accountEntity == null) {
            return null;
        }
        Account account = new Account();
        account.setId(accountEntity.getId());
        account.setCreated(accountEntity.getCreated());
        account.setUpdated(accountEntity.getUpdated());
        account.setUsername(accountEntity.getUsername());
        account.setEmail(accountEntity.getEmail());
        account.setPasswordHash(accountEntity.getPasswordHash());
        account.setPasswordSalt(accountEntity.getPasswordSalt());
        if (accountEntity.getStatus() != null) {
            switch (accountEntity.getStatus()) {
                case active:
                    account.setStatus(Account.Status.active);
                    break;
                case unverified:
                    account.setStatus(Account.Status.unverified);
                    break;
                case disabled:
                    account.setStatus(Account.Status.disabled);
                    break;
                case deleted:
                    account.setStatus(Account.Status.deleted);
                    break;
            }
        }
        return account;
    }
}

File: rust/src/account_mapper.rs
use crate::{
    account::{Account, Status},
    account_entity::{AccountEntity, Status as StatusEntity},
};

impl From<AccountEntity> for Account {
    fn from(account_entity: AccountEntity) -> Account {
        Account {
            id: account_entity.id.into(),
            created: account_entity.created,
            updated: account_entity.updated,
            username: account_entity.username.into(),
            email: account_entity.email.into(),
            password_hash: account_entity.password_hash.into(),
            status: account_entity.status.into(),
        }
    }
}

impl From<StatusEntity> for Status {
    fn from(status: StatusEntity) -> Status {
        match status {
            StatusEntity::Unverified => Status::Unverified,
            StatusEntity::Active => Status::Active,
            StatusEntity::Disabled => Status::Disabled,
            StatusEntity::Deleted => Status::Deleted,
        }
    }
}
